com.unity.netcode.gameobjects/Runtime/Interest/InterestManager.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+namespace Unity.Netcode.Interest
+{
+    internal interface IInterestObject<TObject>
+    {
+        public void AddInterestNode(IInterestNode<TObject> obj);
+        public void RemoveInterestNode(IInterestNode<TObject> obj);
+        public HashSet<IInterestNode<TObject>> GetInterestNodes();
+    }
+
+    // interest *system* instead of interest node ?
+    internal class InterestManager<TObject> where TObject : IInterestObject<TObject>
+    {
+        private readonly InterestNodeStatic<TObject> m_DefaultInterestNode = new InterestNodeStatic<TObject>();
+
+        // Trigger the Interest system to do an update sweep on any Interest nodes
+        //  I am associated with
+        public void UpdateObject(TObject obj)
```

Comment:
Does it make sense to use `ref` within these functions? if TObject happens to be a struct then it will be a copy of the struct are we okay with that?

Reply:
I ref-ified the InterestManager functions.  But I would also want to ref-ify the similar IInterestNode functions, but, having tried that, discovered I can't do that (yet) because I sometimes call things like AddObject() with `this` of a NetworkObject.  This goes back to the improvement we want to later do where TObject is an unmanaged type